### ISSUE TITLE
Problems encountered when splitting the chimeric contigs.

### ISSUE THREAD

**Issue by QingqingLiuaaa** [tanghaibao/jcvi#421]
Why did my genome become smaller after I split the chimeric contigs, it seems that the sequence is missing.

**Comment by tanghaibao** [comment]
@QingqingLiuaaa 

By default, chimeric contigs are split at gaps (contain NNNs). So the genome will appear smaller as a result. It should not remove any real sequences.

If you see real sequences being removed, would you please send the command that you used and some sample input files for me to reproduce the error? Thanks ~

**Comment by QingqingLiuaaa** [comment]
The following is my test file and the command I ran. The ATGC number of genome.fasta is 49166480 bp, while the ATGC number of the genome.SPLIT.fasta file obtained after splitting the chimeric sequence is only 47833389 bp, which means that some real sequence are missing.
[TestData.zip](https://github.com/tanghaibao/jcvi/files/7601717/TestData.zip)

**Comment by tanghaibao** [comment]
@QingqingLiuaaa 

Thank you for sending over the test data. 
You have found a bug in the breakpoint refinement step! It is great that you are checking the size of the assembly before and after the split.

I have corrected this bug but was curious as to why this bug was not found earlier, since I routinely check the sequence size before. To avoid future issues, I just added a test on the refinement step.

Many thanks for this fantastic bug report.

Haibao
